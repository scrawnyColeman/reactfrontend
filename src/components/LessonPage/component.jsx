import React from 'react';
import styled from 'styled-components';
import { useLocation } from 'react-router-dom';
import LessonInformation from '../LessonInformation/component';
import coursedata from '../../data/courses';

const Container = styled.div`
    margin: 20vh 20px 0 20px;
`;

const LessonPage = () => {
    const location = useLocation();
    const lessonId = location.pathname.split('/learn/')[1];
    return (
        <Container>
            <LessonInformation lessonId={lessonId} courses={coursedata.courses} />
        </Container>
    );
};

export default LessonPage;
