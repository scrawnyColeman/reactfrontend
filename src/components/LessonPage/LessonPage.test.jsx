import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import LessonPage from './component';

jest.mock('react-router-dom', () => {
    const useLocation = jest.fn(() => ({ pathname: 'something', search: 'somewhere' }));
    return { useLocation };
});
Enzyme.configure({ adapter: new Adapter() });

describe('LessonPage component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<LessonPage />);
        expect(wrapper).toMatchSnapshot();
    });
});
