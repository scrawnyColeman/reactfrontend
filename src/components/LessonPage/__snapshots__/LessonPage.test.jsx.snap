// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonPage component component matches snapshot 1`] = `""`;
